src/porepy/models/abstract_model.py

```diff
@@ -232,28 +233,42 @@ def _initialize_linear_solver(self) -> None:
         if solver not in ["scipy_sparse", "pypardiso", "umfpack"]:
             raise ValueError(f"Unknown linear solver {solver}")
 
-    def assemble_and_solve_linear_system(self, tol: float) -> np.ndarray:
+    def assemble_linear_system(self):
         """Assemble the linearized system, solve and return the new solution vector.
 
         The linear system is defined by the current state of the model.
 
-        The linear solver is set according to the parameter 'linear_solver' set under
-        initialization of this model.
 
-        Parameters:
-            tol (double): Target tolerance for the linear solver. May be used for
-                inexact approaches.
 
-        Returns:
-            np.ndarray: Solution vector.
+
+
+        Attributes:
+            linear_system is assigned.
 
         """
         t_0 = time.time()
         if self._use_ad:
             A, b = self._eq_manager.assemble()
         else:
             A, b = self.assembler.assemble_matrix_rhs()  # type: ignore
-        t_1 = time.time()
+        self.linear_system = (A, b)
+        logger.debug(f"Assembled linear system in {t_0-time.time():.2e} seconds.")
+
+    def linear_solve(self):
```

Comment:
@pmgbergen/porepy-maint  Although this is largely a matter of taste, I would suggest the more descriptive name `solver_linear_system(self)` instead of `linear_solve(self)`. The suggested name also goes more in line with `assemble_linear_system(self)`.

Reply:
@keileg, could you should weigh in on the naming, please?